src/HealthChecks.DynamoDb/DependencyInjection/DynamoDbHealthCheckBuilderExtensions.cs

```diff
@@ -41,5 +41,39 @@ public static class DynamoDbHealthCheckBuilderExtensions
                 tags,
                 timeout));
         }
+
+        /// <summary>
+        /// Add a health check for AWS DynamoDb database.
+        /// </summary>
+        /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+        /// <param name="connectionOptionsFactory">A factory to build the DynamoDB connection data to use.</param>
+        /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'dynamodb' will be used for the name.</param>
+        /// <param name="failureStatus">
+        /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+        /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+        /// </param>
+        /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+        /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+        /// <returns>The specified <paramref name="builder"/>.</returns>
+        public static IHealthChecksBuilder AddDynamoDb(
+            this IHealthChecksBuilder builder,
+            Func<IServiceProvider, DynamoDBOptions>? connectionOptionsFactory,
```

Comment:
```suggestion
            Func<IServiceProvider, DynamoDBOptions> connectionOptionsFactory,
```

Nullable here makes no sense.